core/tests/blocks.rs

```diff
@@ -0,0 +1,48 @@
+use unimarkup_commons::test_runner::{self, test_file};
+use unimarkup_core::Unimarkup;
+
+pub fn test_block_snapshots() -> Vec<libtest_mimic::Trial> {
+    let tests_path = unimarkup_commons::crate_tests_path!();
+
+    let test_cases = test_runner::collect_tests(
+        tests_path.join("spec/markup"),
+        tests_path.join("spec/snapshots/"),
+        "markup",
+    );
+
+    let mut test_runs = Vec::with_capacity(test_cases.len());
+
+    for case in test_cases {
+        let test_name = format!("{}::{}", module_path!(), case.test.name.as_str());
+
+        let test_run = move || {
+            std::panic::catch_unwind(|| run_test_case(case)).map_err(|err| {
+                let panic_msg = err
+                    .downcast_ref::<&str>()
+                    .unwrap_or(&"Panic message not available");
+
+                format!("Test case panicked: {}", panic_msg).into()
+            })
+        };
+
+        test_runs.push(libtest_mimic::Trial::test(test_name, test_run));
+    }
+
+    test_runs
+}
+
+fn run_test_case(case: test_runner::test_file::TestCase) {
+    test_runner::spec_test::assert_um_spec(
```

Comment:
This asserts that out implementation returned the correct output (currently only HTML).
It is not a snapshot test.
For snapshot testing, you need to add `test_parser_snap!()` from `unimarkup-commons`.

Reply:
I now collect snapshot tests as well. Also the test output now includes that bit of info for each test. 

For example: 

```
test unimarkup_core::runner::spec::simple-paragraph       ... ok
test unimarkup_core::runner::spec::paragraph-with-inlines ... ok
test unimarkup_core::runner::snap::paragraph-with-inlines ... ok
test unimarkup_core::runner::snap::simple-paragraph       ... ok
```